javascript/src/components/grok-patterns/GrokPatterns.jsx

```diff
@@ -61,6 +61,10 @@ var GrokPatterns = React.createClass({
                 </tbody>
             </table>);
     },
+    validPatternName(name) {
+        // Check if patterns already contain a pattern with the given name.
+        return this.state.patterns.map((p) => p.name).indexOf(name) === -1;
+    },
```

Comment:
I'd use a [`some` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) there, as you don't need to go through the whole array when there's a duplicated element.